web/lib/engine/__tests__/questStore.test.ts

```diff
@@ -0,0 +1,90 @@
+import { expect, test } from "vitest";
+import { getNewActionsPerStep } from "../quests";
+import { questStore } from "@/lib/stores/quest-store";
+import {
+	savingsDepositAction,
+	waiterJobAction,
+} from "../actions/standard-actions";
+
+test("add new actions per step", () => {
+	const job = (steps: number) => ({
+		...waiterJobAction,
+		remainingSteps: steps,
+	});
+	const invest = (steps: number, capital: number) => ({
+		...savingsDepositAction,
+		remainingSteps: steps,
+		capital,
+	});
+
+	questStore.send({
+		type: "newActionsAppend",
+		newActions: [],
+	});
+
+	questStore.send({
+		type: "newActionsAppend",
+		newActions: [job(Infinity), invest(2, 1000)],
+	});
+
+	const quest = questStore.get().context;
+
+	const newActionsPerStep = getNewActionsPerStep(quest);
+	expect(newActionsPerStep).toEqual([
+		[],
+		[job(Infinity), invest(2, 1000)],
+		[],
+	]);
+});
+
+test("stop given action", () => {
+	const job = (steps: number) => ({
+		...waiterJobAction,
+		remainingSteps: steps,
+	});
+	const invest = (steps: number, capital: number) => ({
+		...savingsDepositAction,
+		remainingSteps: steps,
+		capital,
+	});
+
+	//TODO: Reset Quest store functionality?
+	//TODO: Investigate possible bug related to the behavior of simulateWithActions() which duplicates the first step actions when currentStepStopAction is called on an action in the first step
```

Comment:
That's my fault - `simulateWithActions()` and `getNewActionsPerStep()` are not complete opposites of each other. Can you add a failing test in the quests test suite that showcases this? You can use `test.fails(...)` to ignore failing test for now.

If you prefer, this can be done in the future